Simplified/MyBooks/NYPLMyBooksDownloadCenter+AudiobooksDownload.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  NYPLMyBooksDownloadCenter+AudiobooksDownload.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-04-06.
+//  Copyright © 2022 NYPL. All rights reserved.
+//
+#if FEATURE_AUDIOBOOKS
+import Foundation
+import NYPLAudiobookToolkit
+
+extension NYPLMyBooksDownloadCenter {
+  /// Create an audiobook object and add to download queue.
+  /// Below are the steps:
+  ///   1. Retrieve book manifest from local storage
+  ///   2. Transform manifest into dictionary and add book identifier if needed
+  ///   3. Update vendor DRM key if needed
+  ///   4. Add audiobook to downloader queue
+  @objc(downloadAudiobookForBook:)
+  func downloadAudiobook(for book: NYPLBook?) {
+    guard let book = book,
+          let url = self.fileURL(forBookIndentifier: book.identifier) else {
+            Log.error(#file, "Missing book or url. Book: \(String(describing: book?.loggableShortString)). URL: \(String(describing:self.fileURL(forBookIndentifier: book?.identifier)))")
+            return
+    }
+    
+    let metadata = [
+      "book": book.loggableShortString(),
+      "fileURL": url.absoluteString
+    ]
+    
+    AudiobookManifestAdapter.transformManifestToDictionary(for: book,
+                                                              fileURL: url) { json, decryptor, error in
+      if error == .none,
+        let json = json {
+        AudioBookVendorsHelper.updateVendorKey(book: json) { [weak self] error in
```

Comment:
@ettore I missed the part which updates the vendor DRM key when refactoring the code from `NYPLBookCellDelegate+Audiobooks`. I have updated the code and added more documentation above.

Reply:
the new changes look good to me. If you have a lot more changes it might be better to do them in a subsequent PR since this one is getting a bit hard to follow.